hydrosdk/model.py

```diff
@@ -366,35 +350,47 @@ def find_by_id(cluster, model_id):
         if resp.ok:
             for model_json in resp.json():
                 if model_json['id'] == model_id:
-                    model_id = model_json["id"]
-                    model_name = model_json["model"]["name"]
-                    model_version = model_json["modelVersion"]
-                    model_contract = contract_from_dict(model_json["modelContract"])
-
-                    model_runtime = DockerImage(model_json["runtime"].get("name"), model_json["runtime"].get("tag"),
-                                                model_json["runtime"].get("sha256"))
-                    model_image = model_json["image"]
-                    model_cluster = cluster
-
-                    res_model = Model(model_id, model_name, model_version, model_contract,
-                                      model_runtime, model_image, model_cluster)
-
-                    return res_model
+                    return Model.from_json(cluster, model_json)
 
         raise Exception(
             f"Failed to find_by_id Model for model_id={model_id}. {resp.status_code} {resp.text}")
 
-    # TODO: method not used
     @staticmethod
-    def from_proto(proto, cluster):
-        Model(
-            id=proto.id,
-            name=proto.name,
-            version=proto.version,
-            contract=proto.contract,
-            runtime=proto.runtime,
-            image=DockerImage(name=proto.image.name, tag=proto.image.tag, sha256=proto.image_sha),
-            cluster=cluster
+    def from_json(cluster, model_json):
+        """
+        Internal method used for deserealization of a Model from json object
+        :param cluster:
+        :param model_json: a dictionary
+        :return: A Model instance
+        """
+        model_id = model_json["id"]
+        model_name = model_json["model"]["name"]
+        model_version = model_json["modelVersion"]
+        model_contract = contract_from_dict(model_json["modelContract"])
+
+        # external model deserialization handling
+        # TODO: get its own endpoint for external model
+        if not model_json.get("runtime"):
+            model_json["runtime"] = {}
+
+        model_runtime = DockerImage(model_json["runtime"].get("name"), model_json["runtime"].get("tag"),
+                                    model_json["runtime"].get("sha256"))
+        model_image = model_json.get("image")
+        model_cluster = cluster
+
+        status = ModelStatus[model_json['status']]
```

Comment:
Request to `/api/v2/model/version/{}/{}` returns this kind of response
```
{'model': {'id': 5, 'name': 'ext-model-test'}, 'modelContract': {'modelName': '', 'predict': {'signatureName': 'infer', 'inputs': [{'profile': 'NONE', 'dtype': 'DT_DOUBLE', 'name': 'in1', 'shape': {'dim': [{'size': -1, 'name': ''}, {'size': 2, 'name': ''}], 'unknownRank': False}}], 'outputs': [{'profile': 'NONE', 'dtype': 'DT_DOUBLE', 'name': 'out1', 'shape': {'dim': [{'size': -1, 'name': ''}], 'unknownRank': False}}]}}, 'id': 10, 'metadata': {'additionalProp1': 'prop'}, 'kind': 'External', 'modelVersion': 3, 'created': '2020-04-13T06:11:16.506Z', 'runtime': {}}
```
There is no status